plugins/elasticsearch/encoders.go

```diff
@@ -326,24 +350,52 @@ func (e *ESJsonEncoder) Encode(pack *PipelinePack) (output []byte, err error) {
 		case "hostname":
 			writeStringField(first, &buf, e.fieldMappings.Hostname, m.GetHostname())
 		case "fields":
-			for _, field := range m.Fields {
-				raw := false
-				if len(e.rawBytesFields) > 0 {
-					for _, raw_field_name := range e.rawBytesFields {
-						if *field.Name == raw_field_name {
-							raw = true
+			if len(e.dynamicFields) > 0 {
+				fmt.Printf("has dynamic fields %s", e.dynamicFields)
+				for _, field := range m.Fields {
```

Comment:
It seems like it might not be uncommon for there to be a large number of fields, with only a few of them being sent to ES. I think it might make more sense, then, to iterate through `e.dynamicFields` and grab all of the field values by name, rather than iterating through all of the fields and then checking to see if each one is in the list.